app/controllers/people_skills_controller.rb

```diff
@@ -2,25 +2,42 @@
 
 class PeopleSkillsController < CrudController
   include ParamConverters
-  helper_method :search_level, :search_interest
 
+  helper_method :search_skill, :search_level, :search_interest, :row_count, :query_params
+
+  # rubocop:disable Metrics/MethodLength
   def entries
     return [] if params[:skill_id].blank?
 
+    extracted_queries = extract_query_params
+    return [] if extracted_queries[:skill_ids].empty?
+
     base = PeopleSkill.includes(:person, skill: [
                                   :category,
                                   :people, { people_skills: :person }
                                 ])
     PeopleSkillsFilter.new(
-      base, true, params[:level], params[:interest], params[:skill_id]
+      base, true, extracted_queries[:levels], extracted_queries[:interests],
+      extracted_queries[:skill_ids]
     ).scope
   end
+  # rubocop:enable Metrics/MethodLength
 
-  def search_level
-    params[:level] ? params[:level].to_i : 1
+  def row_count
+    params[:skill_id].present? ? params[:skill_id].length : 1
   end
 
-  def search_interest
-    params[:interest] ? params[:interest].to_i : 1
+  # rubocop:disable Metrics/AbcSize
+  def extract_query_params
```

Comment:
same for this part ... extract the whole logic how to extract these filter params into an own class that is specific for this task